src/main/kotlin/core/controller/PokemonController.kt

```diff
@@ -0,0 +1,80 @@
+package core.controller
+
+import core.model.PokemonResponse
+import core.model.PokemonResponseMeta
+import core.model.UserPokemon
+import org.springframework.web.bind.annotation.*
+import core.service.PokemonService
+import core.service.UserService
+
+@RestController
+@RequestMapping("/api/v1/pokemon")
+class PokemonController(val userService: UserService, val pokemonService: PokemonService) {
+    @GetMapping
+    fun getPokemon(
+        @RequestParam(value = "search") search: String?,
+        @RequestParam(value = "perPage", defaultValue = "20") perPage: Int,
+        @RequestParam(value = "page", defaultValue = "1") page: Int,
+        @RequestParam(value = "captured") captured: String?,
+        @RequestHeader("Authorization") authorization: String
+    ): PokemonResponse {
+        val user = userService.validateUser(authorization)
+        var pokemon = pokemonService.findPokemon(user)
+
+        if (captured?.toLowerCase() == "true") {
+            pokemon = pokemon.filter { it.captured }
+        } else if (captured?.toLowerCase() == "false") {
+            pokemon = pokemon.filter { !it.captured }
+        }
+
+        search?.let {
+            pokemon = pokemon.filter {
+                it.name?.toLowerCase()?.contains(search?.toLowerCase() ?: "") ?: false
+            }
+        }
+
+        var paginatedPokemon = listOf<UserPokemon>()
+        val start = perPage * (page - 1)
+        val end = Math.min(perPage * page, pokemon.count())
+        if (start <= pokemon.count() && start >= 0) {
+            paginatedPokemon = pokemon.subList(start, end)
+        }
+
+        return PokemonResponse(
+            paginatedPokemon,
+            PokemonResponseMeta(
+                pokemon.count(),
+                page,
+                Math.ceil(pokemon.count().toDouble() / perPage).toInt(),
+                perPage
+            )
+        )
+    }
+
+    @GetMapping(value = ["/{id}"])
+    fun getPokemon(
+        @PathVariable id: Int,
```

Comment:
Why are you taking in the `id` as an integer on these routes if you always convert it to a string before using it? 

Taking it in as an `Int` causes a conversion from the `string` url, and then you do the conversion back, so it should save a tiny amount of time to leave it as a string.